### ISSUE TITLE
Matching Heuristic causes Mismatches, e.g. "Deep Learning". Check for >=1 same author?

### ISSUE THREAD

**Issue by lgalke** [yuchenlin/rebiber#50]
Big thanks to the people behind rebiber. It is a very helpful tool.

I noticed some peculiarities with the matching heuristic (see example below).
TLDR: rebiber turned the entries for the Deep Learning book (Goodfellow) and the Deep Learning nature article (LeCun, Bengio, Hinton) into entries referring to a SIGKDD paper (Salakuthdinov) with the same title--without any warning.

Based on this behaviour, I assume the matching is done only based on title.

I suggest to check additionally for *at least one common author*. Or (less invasive): to emit an adequate warning when there is not a single same-author-name in a new entry compared to the original entry.

## Input 

```
@article{deeplearning,
  title = {Deep Learning},
  author = {LeCun, Yann and Bengio, Yoshua and Hinton, Geoffrey},
  year = {2015},
  month = may,
  journal = {Nature},
  volume = {521},
  number = {7553},
  pages = {436--444},
  publisher = {{Nature Publishing Group}},
  issn = {1476-4687},
  doi = {10.1038/nature14539},
  abstract = {Deep learning allows computational models that are composed of multiple processing layers to learn representations of data with multiple levels of abstraction. These methods have dramatically improved the state-of-the-art in speech recognition, visual object recognition, object detection and many other domains such as drug discovery and genomics. Deep learning discovers intricate structure in large data sets by using the backpropagation algorithm to indicate how a machine should change its internal parameters that are used to compute the representation in each layer from the representation in the previous layer. Deep convolutional nets have brought about breakthroughs in processing images, video, speech and audio, whereas recurrent nets have shone light on sequential data such as text and speech.},
  copyright = {2015 Nature Publishing Group, a division of Macmillan Publishers Limited. All Rights Reserved.},
  langid = {english},
  keywords = {Computer science,Mathematics and computing},
  file = {/Users/lukas/Zotero/storage/MTC45LC9/LeCun et al. - 2015 - Deep learning.pdf}
}

@inproceedings{deeplearning,
 author = {Ruslan Salakhutdinov},
 bibsource = {dblp computer science bibliography, https://dblp.org},
 biburl = {https://dblp.org/rec/conf/kdd/Salakhutdinov14.bib},
 booktitle = {Proc. of KDD},
 doi = {10.1145/2623330.2630809},
 pages = {1973},
 publisher = {{ACM}},
 timestamp = {Tue, 06 Nov 2018 00:00:00 +0100},
 title = {Deep learning},
 url = {https://doi.org/10.1145/2623330.2630809},
 year = {2014}
}
```

Cmd: `rebiber -i main.bib -o main.bib -r editor -d True -s True`

## Output

```
@inproceedings{deeplearning,
 author = {Ruslan Salakhutdinov},
 bibsource = {dblp computer science bibliography, https://dblp.org},
 biburl = {https://dblp.org/rec/conf/kdd/Salakhutdinov14.bib},
 booktitle = {Proc. of KDD},
 doi = {10.1145/2623330.2630809},
 pages = {1973},
 publisher = {{ACM}},
 timestamp = {Tue, 06 Nov 2018 00:00:00 +0100},
 title = {Deep learning},
 url = {https://doi.org/10.1145/2623330.2630809},
 year = {2014}
}

@inproceedings{goodfellow_deep_2016,
 author = {Ruslan Salakhutdinov},
 bibsource = {dblp computer science bibliography, https://dblp.org},
 biburl = {https://dblp.org/rec/conf/kdd/Salakhutdinov14.bib},
 booktitle = {Proc. of KDD},
 doi = {10.1145/2623330.2630809},
 pages = {1973},
 publisher = {{ACM}},
 timestamp = {Tue, 06 Nov 2018 00:00:00 +0100},
 title = {Deep learning},
 url = {https://doi.org/10.1145/2623330.2630809},
 year = {2014}
}
```

The command line output of rebiber only state
- `Converted. ID: deeplearning ; Title: Deep Learning`
-  `Converted. ID: goodfellow_deep_2016 ; Title: Deep {Learning}`

## Desired output

a) Do not have the entries replaced, as there is not a single author name in common

or b) Emit a major warning on the command line, when replacing without a single author name in common

**Comment by yuchenlin** [comment]
Hey @lgalke  thanks a lot for reporting this error and the suggestions. I feel like the waning is an easier option and we can let the users to deal with this manually. However, I'm not having enough bandwidth recently. If you'd like to create a PR to output a warning message, it would be much helpful! Thank you very much!
